packages/extension-ui/src/Popup/Accounts/index.tsx

```diff
@@ -60,24 +63,30 @@ function Accounts({ className }: Props): React.ReactElement {
           />
           <ScrollWrapper>
             <div className={className}>
-              {Object.entries(groupedParents).map(([networkName, details]) => (
-                <div key={networkName}>
-                  {networkName !== defaultNetwork && <span className='network-heading'>{networkName}</span>}
-                  {details.map((json) => (
-                    <AccountsTree
-                      {...json}
-                      key={json.address}
-                    />
-                  ))}
-                  {filterChildren(networkName, defaultNetwork, details).map((json) => (
-                    <AccountsTree
-                      {...json}
-                      key={json.address}
-                      parentName={getParentName(json)}
-                    />
-                  ))}
-                </div>
-              ))}
+              {Object.entries(groupedParents).map(([networkName, details]) => {
+                if (details.length > 0) {
+                  return (
+                    <div key={networkName}>
+                      {networkName !== defaultNetwork && <span className='network-heading'>{networkName}</span>}
+                      {details.map((json) => (
+                        <AccountsTree
+                          {...json}
+                          key={json.address}
+                        />
+                      ))}
+                      {filterChildren(networkName, defaultNetwork, details).map((json) => (
+                        <AccountsTree
+                          {...json}
+                          key={json.address}
+                          parentName={getParentName(json)}
+                        />
+                      ))}
+                    </div>
+                  );
+                } else {
+                  return null;
+                }
+              })}
```

Comment:
It get's pretty big - maybe we can take it outside of template and just use const in the template?